DQM/HLTEvF/plugins/LumiMonitor.cc

```diff
@@ -0,0 +1,246 @@
+#include "DQM/HLTEvF/plugins/LumiMonitor.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DQM/TrackingMonitor/interface/GetLumi.h"
+
+
+// -----------------------------
+//  constructors and destructor
+// -----------------------------
+
+LumiMonitor::LumiMonitor( const edm::ParameterSet& iConfig ) : 
+  folderName_                     ( iConfig.getParameter<std::string>("FolderName")                  )
+  , pixelClustersToken_           ( mayConsume<edmNew::DetSetVector<SiPixelCluster> >(iConfig.getParameter<edm::InputTag>("pixelClusters") ) )
```

Comment:
instead of using mayConsume,  I would suggest to
  - move doPixelLumi_ (et al.) before pixelClustersToken_
  - initialise pixelClustersToken_ based on the value of doPixelLumi_:

```
pixelClustersToken_( doPixelLumi_ ? consumes<...>(...) : edm::EDGetTokenT<edmNew::DetSetVector<SiPixelCluster>> () )
```